### ISSUE TITLE
Add toUpper and toLower methods

### ISSUE THREAD

**Issue by pasamio** [stedolan/jq#492]
Looking through the system, support for a toUpper and toLower might make some folks lives easier. In my case I have some keys that are upper case, some that are mixed case and others that are lower case and I'd like to use a single match on them by being able to use the same case.

**Comment by nikolay** [comment]
:+1: I really need this as well!

**Comment by pkoppstein** [comment]
@nikolay - ascii_upcase and ascii_downcase have been in "master" on github since Dec 27.  Their definitions can also be used in jq 1.4, as explained in the [FAQ](/stedolan/jq/wiki/FAQ).

**Comment by gabrielmagno** [comment]
Any plans to implement "upcase" and "downcase" functions with UTF-8 support, something similar to [awk's "tolower"](https://www.gnu.org/software/gawk/manual/html_node/String-Functions.html).

With jq 1.5: ```echo '"ÁgUA"' | jq '. | ascii_downcase'```
Output: ```"Água"```

With GNU Awk 4.1.3: ```echo '"ÁgUA"' | awk '{print(tolower($0))}'```
Output: ```"água"```

**Comment by paulochf** [comment]
This :point_up: would be great!

**Comment by AndresPineros** [comment]
Currently I have to create multiple jq queries and link them or use linux commands with regular expressions to lower case only the target fields

**Comment by eintr** [comment]
Waiting for this one.

**Comment by flaviotordini** [comment]
Started today using jq and found myself needing this.

**Comment by LordMike** [comment]
:+1:

**Comment by mmacek7** [comment]
My wish would be to support regexp substitions from perlre:
\l \u \L \U \E

**Comment by dsynkd** [comment]
doesn't `ascii_downcase` from `builtin.jq` do this?

**Comment by gabrielmagno** [comment]
> I found `ascii_downcase` from `builtin.jq` to do just this

Yes, but as the name suggests, it works only for ASCII characters. For instance, UTF-8 characters of accented letters (e.g. `"Á"` or `"Ñ"`) in a string, will be simply ignored and returned in uppercase.

The current workaround, as stated in [my 2016 comment in this issue](https://github.com/stedolan/jq/issues/492#issuecomment-199442775), is using awk's `tolower`, wich properly supports accented letters.

I hope that something similar will be implemented in jq someday.

**Comment by liquidaty** [comment]
seems that utf8 case folding should be fairly straightforward using utf8proc (https://juliastrings.github.io/utf8proc/doc/). We might give it a go, anyone interested to take a look if we do?

**Comment by liquidaty** [comment]
I've implemented this as a couple of built-ins, `utf8tolower` and `utf8toupper`. It works like so:
```
> echo '"ǄIǄA"' | jq '.|utf8tolower'
"ǆiǆa"
```

The code changes are small and easily toggled, though . Is there any interest from this repo's maintainers to incorporate these changes? If so pls lmk and I will generate a pull request.

**Comment by liquidaty** [comment]
@nicowilliams @gabrielmagno @flaviotordini @eintr @paulochf @LordMike: given that it seems from your comments (years ago by now...) that you'd have an interest in a solution-- any thoughts on the above?

**Comment by liquidaty** [comment]
@gabrielmagno, anyone else: here's a diff from the latest jq version pulled today: https://github.com/liquidaty/jq/commit/9f0bde5f5a560e021bc05be98906a0e555270f4e

Note that the autoconf includes settings that appear to support building utf8proc, but that part doesn't actually work-- rather, it just looks for utf8proc already being available and uses it if found

**Comment by liquidaty** [comment]
Pull request: https://github.com/stedolan/jq/pull/2547
